lib/vector-core/src/metrics/handle.rs

```diff
@@ -283,6 +289,32 @@ mod test {
             .quickcheck(inner as fn(Vec<f64>) -> TestResult);
     }
 
+    #[test]
+    fn historgram_uses_right_buckets() {
```

Comment:
I'll be a little annoying here: we should have a property test for this.

Probably extract out the `let bucket = ...` logic into a function, and then have the property test try and show that we never return anything other than `0 => bucket_idx => Self::BUCKETS - 1`.

If it was pure integer math, I'd say: screw it, the invariants are simple enough to test by hand. With floating-point numbers, I'm a little more hesitant, especially with all the various infinities, NaNs, etc.

Reply:
Agreed and, with the calculation broken out, that becomes much easier. Done in 27f99ce.